riscv/tests/riscv_data/poseidon_gl_via_coprocessor.rs

```diff
@@ -0,0 +1,55 @@
+#![no_std]
+
+use runtime::coprocessors::poseidon_gl;
+
+#[no_mangle]
+fn main() {
+    let mut i: [u64; 12] = [0; 12];
+    poseidon_gl(&mut i);
+    assert_eq!(i[0], 4330397376401421145);
+    assert_eq!(i[1], 14124799381142128323);
+    assert_eq!(i[2], 8742572140681234676);
+    assert_eq!(i[3], 14345658006221440202);
+
+    let mut i: [u64; 12] = [1; 12];
+    poseidon_gl(&mut i);
+    assert_eq!(i[0], 16428316519797902711);
+    assert_eq!(i[1], 13351830238340666928);
+    assert_eq!(i[2], 682362844289978626);
+    assert_eq!(i[3], 12150588177266359240);
+
+    let minus_one = 0xffffffff00000001 - 1;
+    let mut i: [u64; 12] = [minus_one; 12];
+    poseidon_gl(&mut i);
+    assert_eq!(i[0], 13691089994624172887);
+    assert_eq!(i[1], 15662102337790434313);
+    assert_eq!(i[2], 14940024623104903507);
+    assert_eq!(i[3], 10772674582659927682);
+
+    let mut i: [u64; 12] = [18446744069414584321; 12];
```

Comment:
The original test sets the capacity (last 4 elements) to 0. How can the test still pass?

If I change set the capacity to 0 the test still passes, so looks like the capacity is always set to 0.

Reply:
that's indeed weird, need to take a look